lib/commands.js

```diff
@@ -522,7 +522,18 @@ var operations = module.exports.operations = {
 
   createPreamble: function(context) {
     var preamble = d3.queue();
-    preamble.defer(template.read, context.templatePath, context.overrides.templateOptions);
+
+    if (Object.prototype.toString.call(context.template) === "[object String]") {
```

Comment:
I'm curious why you went this way with the condition instead of one of these other options. Does this cover some edge case I'm overlooking?

```
if (context.template.toString() === '[object String]')
```

```
if (typeof context.template === 'string')
```